num/quat/slerp_example_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quat_test
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/num/quat"
+	"gonum.org/v1/gonum/spatial/r3"
+)
+
+// slerp returns the spherical interpolation between q0 and q1
+// for t in [0,1].
+func slerp(r0, r1 r3.Rotation, t float64) r3.Rotation {
+	q0 := quat.Number(r0)
+	q1 := quat.Number(r1)
+	// Simo Särkkä "Notes on Quaternions", June 28, 2007
+	q1 = quat.Mul(q1, quat.Inv(q0))
+	return r3.Rotation(quat.Mul(quat.Exp(quat.Scale(t, quat.Log(q1))), q0))
+}
+
+// Spherically interpolate between two quaternions to obtain a rotation.
+func Example_slerp() {
+	const nsamples = 10
+	noRot := r3.NewRotation(0, r3.Vec{})
```

Comment:
Given that the base case where there is no initial rotation is a lot simpler than when there is, it would be good to show that this does something. So I'd suggest having the initial rotation be pi/4 around x.

Also rename `noRot` => `rot0` and `xRot` => `rot1`.